src/lightning/data/datasets/base.py

```diff
@@ -0,0 +1,37 @@
+from typing import Any
+
+from torch.utils.data import Dataset as TorchDataset
+
+from lightning.data.backends import _DatasetBackend, LocalDatasetBackend, S3DatasetBackend
+from lightning.data.fileio import OpenCloudFileObj
+
+
+class _Dataset(TorchDataset):
+    """Base dataset class for streaming data from a cloud storage.
+
+    Args:
+        backend: storage location of the data_source. current options are "s3" or "local"
+    """
+
+    def __init__(self, backend: str = "local"):
+        self.backend = self._init_backend(backend=backend)
+
+        assert isinstance(self.backend, _DatasetBackend)
+
+    def _init_backend(self, backend: str) -> _DatasetBackend:
+        """Picks the correct backend handler."""
+        if backend == "s3":
+            return S3DatasetBackend()
```

Comment:
the base class is just protected so that nobody tries to use it directly. This was previously somewhere else, just moved it to this class for code sharing. Left everything else as is, won't touch it now :D